src/o365/yammer/commands/user/user-list.ts

```diff
@@ -0,0 +1,195 @@
+        if (res.users)
+          userOutput = res.users;
+     
+        if (args.options.output === 'json') {
+          this.items = this.items.concat(userOutput);
+        }
+        else {
+          this.items = this.items.concat((userOutput as any[]).map(n => {
+            const item: any = {
+              id: n.id, 
+              full_name: n.full_name, 
+              email: n.email, 
+              job_title: n.job_title, 
+              state: n.state, 
+              url: n.url
+            };
+            return item;
+          }));
+        }
+
+        // this is executed once at the end if the limit operation has been executed
+        // we need to return the array of the desired size. The API does not provide such a feature
+        if (args.options.limit !== undefined && this.items.length > args.options.limit) {
+          this.items = this.items.slice(0,args.options.limit);
+          resolve();
+        }
+        else {
+          // if the groups endpoint is used, the more_available will tell if a new retrieval is required
+          // if the user endpoint is used, we need to page by 50 items (hardcoded)
+          if (res.more_available === true || this.items.length % 50 === 0) {
+            this.getAllItems(cmd, args, ++page)
+                .then((): void => {
+                  resolve();
+                }, (err: any): void => {
+                  reject(err);
+                });
+          }
+          else {
+            resolve();
+          }
+        }
+      }, (err: any): void => {
+        reject(err);
+      });
+    });
+  };
+
+  public commandAction(cmd: CommandInstance, args: CommandArgs, cb: () => void): void {
+    this
+    .getAllItems(cmd, args, 1)
+    .then((): void => {
+        cmd.log(this.items);
+        cb();
+    }, (err: any): void => this.handleRejectedODataJsonPromise(err, cmd, cb));
+  };
+
+  public options(): CommandOption[] {
+    const options: CommandOption[] = [
+      {
+        option: '-g, --groupId [groupId]',
+        description: 'Returns users within a given group'
+      },
+      {
+        option: '-l, --letter [letter]',
+        description: 'Returns users with usernames beginning with the given character'
+      },
+      {
+        option: '--reverse',
+        description: 'Returns users in reverse sorting order'
+      },
+      {
+        option: '--limit [limit]',
+        description: 'Limits the users returned'
+      },
+      {
+        option: '--sortBy [sortBy]',
+        description: 'Returns users sorted by number of messages or followers, instead of the default behavior of sorting alphabetically',
+        autocomplete: ['messages', 'followers']
+      }
+    ];
+
+    const parentOptions: CommandOption[] = super.options();
+    return options.concat(parentOptions);
+  }
+
+  public validate(): CommandValidate {
+    return (args: CommandArgs): boolean | string => {
+      return true;
```

Comment:
We're missing the following validation:

- letter is a letter and just one character
- limit is a number
- sortBy contains correct value